### DISCUSSION TITLE
Asynchronous loading with MultiplayerSpawner

### DISCUSSION THREAD

**Discussion by masterthdev** [godotengine/godot-proposals#11247] [Engine Core]
As far as I understand, there is no way to show a loading screen when using high level multiplayer. I tried to find a workaround but none of these worked:
- Spawning a loading screen instead of level. This leads to “node not found” errors when host loads scene faster than clients.
- Preloading all levels to memory. This wastes memory and takes too long on my project.
- Using rpcs to make sure all peers have scene loaded. This is only suitable for lobby games. It makes everyone wait for no reason, and doesn’t work when a player joins the game later.
- Running MultiplayerSpawner in a thread group. This doesn’t make any difference.
- Disabling public visibility, and manually enabling it when a peer loads the scene. This causes "cache not found" errors and breaks nested spawners, they cannot despawn anything.

MultiplayerSpawner uses ResourceLoader.load() which hangs the main thread. I think MultiplayerSpawner needs some mechanism to load scenes asynchronously, and nested MultiplayerSpawner and MultiplayerSynchronizers should wait until client loads the parent node.

**Comment by jtylerroth** [comment]
I've spent the last 2 days trying to figure out a solution to this. I've come to the same conclusion. The high level options we have just don't support any kind of loading screen when using MultiplayerSpawner. If you've found something let me know!

**Reply by masterthdev** [reply]
I think I tried everything in high level api except using authentication system. Maybe I can send scene list with send_auth, so I can load scenes asynchronously in auth_callback in client before completing authentication. This may work with a long auth_timeout but I don't know if it is secure or not. Also I don't think this will work with interest management, it will probably cause same "cache not found" errors I had with other methods.
